pkg/resourceinterpreter/defaultinterpreter/aggregatestatus_test.go

```diff
@@ -0,0 +1,335 @@
+	oldObj, _ := helper.ToUnstructured(oldDeploy)
+	newObj, _ := helper.ToUnstructured(newDeploy)
+
+	tests := []struct {
+		oldObj                *unstructured.Unstructured
+		aggregatedStatusItems []workv1alpha2.AggregatedStatusItem
+		newObj                *unstructured.Unstructured
+		wantErr               error
+	}{
+		{
+			oldObj:                oldObj,
+			aggregatedStatusItems: aggregatedStatusItems,
+			newObj:                newObj,
+			wantErr:               nil,
+		},
+		{
+			oldObj:                newObj,
+			aggregatedStatusItems: aggregatedStatusItems,
+			newObj:                newObj,
+			wantErr:               nil,
+		},
+	}
+
+	for _, tt := range tests {
+		actualObj, actualErr := aggregateDeploymentStatus(tt.oldObj, tt.aggregatedStatusItems)
+		assert.Equal(t, tt.newObj, actualObj)
+		assert.Equal(t, tt.wantErr, actualErr)
+	}
+}
+
+func TestAggregateServiceStatus(t *testing.T) {
+	statusMapNotLB := map[string]interface{}{}
+	rawNotLB, _ := helper.BuildStatusRawExtension(statusMapNotLB)
+	aggregatedStatusItemsNotLB := []workv1alpha2.AggregatedStatusItem{
+		{ClusterName: "member1", Status: rawNotLB, Applied: true},
+	}
+
+	statusMapLB := map[string]interface{}{
+		"loadBalancer": corev1.LoadBalancerStatus{Ingress: []corev1.LoadBalancerIngress{{IP: "8.8.8.8"}}},
+	}
+	rawLB, _ := helper.BuildStatusRawExtension(statusMapLB)
+	aggregatedStatusItemsLB := []workv1alpha2.AggregatedStatusItem{
+		{ClusterName: "member1", Status: rawLB, Applied: true},
+	}
+
+	serviceClusterIP := &corev1.Service{Spec: corev1.ServiceSpec{Type: corev1.ServiceTypeClusterIP}}
+	serviceNodePort := &corev1.Service{Spec: corev1.ServiceSpec{Type: corev1.ServiceTypeNodePort}}
+	serviceExternalName := &corev1.Service{Spec: corev1.ServiceSpec{Type: corev1.ServiceTypeExternalName}}
+	objServiceClusterIP, _ := helper.ToUnstructured(serviceClusterIP)
+	objServiceNodePort, _ := helper.ToUnstructured(serviceNodePort)
+	objServiceExternalName, _ := helper.ToUnstructured(serviceExternalName)
+
+	oldServiceLoadBalancer := &corev1.Service{Spec: corev1.ServiceSpec{Type: corev1.ServiceTypeLoadBalancer}}
+	newServiceLoadBalancer := &corev1.Service{Spec: corev1.ServiceSpec{Type: corev1.ServiceTypeLoadBalancer}, Status: corev1.ServiceStatus{LoadBalancer: corev1.LoadBalancerStatus{Ingress: []corev1.LoadBalancerIngress{{IP: "8.8.8.8", Hostname: "member1"}}}}}
+	oldObjServiceLoadBalancer, _ := helper.ToUnstructured(oldServiceLoadBalancer)
+	newObjServiceLoadBalancer, _ := helper.ToUnstructured(newServiceLoadBalancer)
+
+	tests := []struct {
+		oldObj                *unstructured.Unstructured
+		aggregatedStatusItems []workv1alpha2.AggregatedStatusItem
+		newObj                *unstructured.Unstructured
+		wantErr               error
+	}{
+		{
+			oldObj:                objServiceClusterIP,
+			aggregatedStatusItems: aggregatedStatusItemsNotLB,
+			newObj:                objServiceClusterIP,
+			wantErr:               nil,
```

Comment:
Without a test case name, the purpose of the test is not clear at a glance.

Reply:
Thanks for advice. Add testcase name.